### PR TITLE
Autofs fix

### PR DESCRIPTION
**Pull request by cclerget**
**Description of the Pull Request (PR):**

- Commit a44d1932631e137326b280cae94cbdc0e107d4a4 address issues #714, #757, it opens image file/directory before entering in mount namespace
- Commit 445152390173becaa7a1b3ccaaf76bcad7a69bff address issues #757, #347 and adds a configuration option "autofs bug path" in singularity.conf which allow admins to add autofs mount directory generating **"Too many levels of symbolic links"** (ex: for bind path or user bind path). autofs option timeout=0 is not required anymore

**This fixes or addresses the following GitHub issues:**

- Ref: #714, #347, #757

**Checkoff for all PRs:**

- [x] I have read the [Guidelines for Contributing](https://github.com/singularityware/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- [x] I have tested this PR locally with a `make test`
- [ ] This PR is NOT against the project's master branch
- [x] I have added myself as a contributor to the [Author's file](https://github.com/singularityware/singularity/blob/master/AUTHORS.md)
- [x] This PR is ready for review and/or merge

Attn: @singularityware-admin

### CODE REVIEW

**Comment by vsoch** [comment]
hey @cclerget could we do this PR against development branch?

File: src/lib/image/open/open.c [hunk 1]
```diff
@@ -60,12 +60,10 @@ int _singularity_image_open(struct image_object *image, int open_flags) {
     }
 
     singularity_message(DEBUG, "Opening file descriptor to image: %s\n", image->path);
-    if ( ( image->fd = open(image->path, open_flags, 0755) ) < 0 ) {
+    if ( ( image->fd = open(image->path, open_flags|O_CLOEXEC, 0755) ) < 0 ) {
```
Comment (gmkurtzer) [inline src/lib/image/open/open.c]: @cclerget, we have had problems in the past with SuSE not supporting this option to open(). Instead could we use fcntl()?

**Comment by gmkurtzer** [comment]
@cclerget Is it possible to do these opens via the various bind functions so it happens automatically (instead of having to call a fixing function that relies on a config entry)?

Thanks, and great thinking!

File: src/lib/image/open/open.c (hunk 1 above)
Comment (cclerget) [reply src/lib/image/open/open.c]: Hi @gmkurtzer , I was asking why you used fcntl ! no problem to add fcntl

**Comment by cclerget** [comment]
Hi @vsoch, of course, I will pull on dev branch

@gmkurtzer open must happens before entering mount namespace to work as expected, so open in bind functions has no effect for the bug.

At first I wrote autofs function which parsed bind path, userbinds, WORKDIR, SCRATCHDIR and automatically open dir/files when a boolean option in configuration file "autofs bug = yes" is enabled. There are drawbacks with this approach, if userbinds or bind path syntax change it imply to change code twice, same thing if a singularity env DIR is added. Or potentially reach file descriptor limit.

So the option to put buggy paths in configuration file should give admins more liberty, for devs less work

**Comment by gmkurtzer** [comment]
@cclerget Ohhh, I get it now, and I understand why this was a problem now! Great catch :)

We have some conflicts with the development branch that need resolution.

**Comment by cclerget** [comment]
@gmkurtzer should be ok for merge

**Comment by gmkurtzer** [comment]
Looks great, thank you @cclerget!

**Pull request merged**